snap/snapcraft.yaml

```diff
@@ -95,6 +95,11 @@ parts:
     source: http://ftp.wayne.edu/apache/httpd/httpd-2.4.33.tar.bz2
     source-checksum: sha256/de02511859b00d17845b9abdd1f975d5ccb5d0b280c567da5bf2ad4b70846f05
 
+    build-packages:
+    - libnghttp2-14
+    - libnghttp2-dev
+    - nghttp2
```

Comment:
Those packages seem to be taken straight from Debian, I have no idea why Ubuntu is not updating them, the [source package](https://packages.debian.org/source/buster/nghttp2) is at `1.32.0`, which you named as the latest release.

I know quite a bit about deb-packaging, maybe helping 'Ubuntu MOTU Developers' to port this from `buster` would be a better use of our time than implementing building nghttp2 and its libs from source? But I would need to investigate how I could go forward adopting this package.

Reply:
Wow @r4co0n, if we could do that backporting it would be awesome! Definitely a better option that what I proposed...the only thing that I don't know is what it takes to convince the "MOTU developers" to accept an updated package in universe.

Reply:
I'm afraid it doesn't really work that way, guys. This is actually one of the reasons snaps exist: once an Ubuntu release is cut, the only way to get updates into its repos is with a [Stable Release Update](https://wiki.ubuntu.com/StableReleaseUpdates) (SRU), which essentially requires the change to be a bugfix (read the doc for more details).

Ubuntu only syncs from Debian when a release is being developed. A bit before the release is cut, syncs are halted. This is why Xenial only has for example v1.7.1 of nghttp2 (note that Bionic has v1.30). Since today we still need to build snaps on Xenial, we're stuck either using v1.7.1, or building from source. No amount of Debian packaging will help us.

Again, this is one of the reasons why snaps exist: since they are isolated and bundle their dependencies, they provide a way for publishers to update their own software on old (in some cases quite old) releases without going through the SRU process (because there's no way they can break other software on the system).

Reply:
Ok, fair enough. So, for this specific case @kyrofa : do you think it's worthwhile pulling the source code and compile it for the snap instead of using the packages?

Reply:
@pachulo I'm afraid I'm not quite confident in my knowledge on this topic or those dependencies to have much of an opinion. I still want to investigate this topic, so perhaps it's best to keep things simple here for now until we're able to properly learn about it.